Add documentation for dynamic kubelet config

phiphi282 opened the pull request:
Add short documentation for how to enable the feature and where to edit
the kubelet config.
Add some links to kubernetes documentation for further references.

I am open to suggestions and changes :)

phiphi282 committed:
Add documentation for dynamic kubelet config.

Add short documentation for how to enable the feature and where to edit
the kubelet config.
Add some links to kubernetes documentation for further references.

phiphi282 committed:
Add newline to end of file

eduardostalinho commented on pages/02.documentation/20.dynamic-kubelet-config/default.en.md:
```diff
@@ -0,0 +1,36 @@
+
+! misconfiguring the kubelet config can possibly prevent the kubelet from starting up.
+
+When this option is enabled multiple configmaps for all supported kubernetes versions will be created in the `kube-system` namespace.
```
It's good to notice the user about the names of the configmaps:
When this option is enabled, multiple configmaps for all supported kubernetes version will be created with the names `kubelet-config-<kubernetes-version>` in the kube-system namespace.
```suggestion
When this option is enabled, multiple configmaps for all supported kubernetes version will be created with the names `kubelet-config-<kubernetes-minor-version>` in the kube-system namespace.
```

phiphi282 replied:
Changed it up in a slightly different way :)

eduardostalinho commented on pages/02.documentation/20.dynamic-kubelet-config/default.en.md:
```diff
@@ -0,0 +1,36 @@
+
+When this option is enabled multiple configmaps for all supported kubernetes versions will be created in the `kube-system` namespace.
+
+All nodes are watching for changes in the configmap with the same kubernetes version that is running on the node.
```
```suggestion
All nodes for the node-deployment are watching for changes in the configmap with the same kubernetes version that is running on the node.
```

eduardostalinho commented on pages/02.documentation/20.dynamic-kubelet-config/default.en.md:
```diff
@@ -0,0 +1,36 @@
+
+Once a change is noticed `systemd` will restart the kubelet so that it can use the new configuration.
+
+All customizable kubelet configuration can be found in the [Kubernetes Docs](https://godoc.org/k8s.io/kubelet/config/v1beta1#KubeletConfiguration).
```
This paragraph is better at the end of the documentation. Because I think that the user would like to read about the kubelet configuration in a broader sense and then see what are the possible configuration values.

eduardostalinho commented on pages/02.documentation/20.dynamic-kubelet-config/default.en.md:
```diff
@@ -0,0 +1,36 @@
+Once a change is noticed `systemd` will restart the kubelet so that it can use the new configuration.
+
+All customizable kubelet configuration can be found in the [Kubernetes Docs](https://godoc.org/k8s.io/kubelet/config/v1beta1#KubeletConfiguration).
+The config keys that are being used in the kubelet configuration are the names after the `json:` prefix.
```
It would be nice to show what are the configuration values in an easier way. Pointing that the configmap can work as an example.

phiphi282 replied:
Not exactly sure what you mean here. Should I paste one of our default configurations in here?

eduardostalinho replied:
Yes.

phiphi282 committed:
Update pages/02.documentation/20.dynamic-kubelet-config/default.en.md

Clarify which nodes are watching for changes for dynamic kubelet config

Co-authored-by: Eduardo "Stalinho" Oliveira de Carvalho <eduardooc.86@gmail.com>

phiphi282 committed:
Apply suggestions

Rephrase where the configmaps are being created.
Change order of documents for further reference.

phiphi282 committed:
Add trailing newline

phiphi282 committed:
Added example configmap for kubernetes 1.17

Changed order of further reference again because it makes more sense to
show link to all configuration options right after example.

phiphi282 committed:
Fix linting errors
